lib/base.js

```diff
@@ -120,8 +120,12 @@ class BaseObject extends CallableObject {
    * @param {callback} cb - The callback that handle the response
    */
   patch(options, cb) {
-    this.api.patch({ path: this._path(options), body: options.body },
-                   cb200(cb));
+    const patchOptions = {
+      path: this._path(options),
+      body: options.body
+    };
+    if ('headers' in options) patchOptions.headers = options.headers;
```

Comment:
I don't think this check is needed... The request options here will grab headers regardless if they exist or not
https://github.com/godaddy/kubernetes-client/blob/a4708cc24f6d29930d32aa6d12b4e07885c42765/lib/api-group.js#L109-L116
Might as well just assign the value without checking for the property.